### PR TITLE
Add incubating messages for VFS retention and partial invalidation

### PR DESCRIPTION
**Pull request by wolfs**
So it is easier to see if the system properties are picked up.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java [hunk 1]
```diff
@@ -174,12 +175,16 @@ VirtualFileSystem createVirtualFileSystem(
                     public void afterStart(GradleInternal gradle) {
                         Map<String, String> systemPropertiesArgs = gradle.getStartParameter().getSystemPropertiesArgs();
                         if (isRetentionEnabled(systemPropertiesArgs)) {
+                            DeprecationLogger.incubatingFeatureUsed("Virtual File System Retention");
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Should we refer to `SingleMessageLogger` instead? I mean, this is not a deprecation, so...

File: subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy [hunk 2]
```diff
@@ -174,6 +175,40 @@ class VirtualFileSystemRetentionIntegrationTest extends AbstractIntegrationSpec
...
+        then:
+        outputContains(incubatingMessage)
+
+        when:
+        run("assemble")
+        then:
+        outputDoesNotContain(incubatingMessage)
+    }
+
+    def "incubating message is shown for partial invalidation"() {
```
Comment (lptr) [inline subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy]: Should this be in a separate test class for partial invalidation instead?

File: subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java (hunk 1 above)
Comment (lptr) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: I think we shouldn't use title case here.

Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java]: Done.

File: subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy (hunk 2 above)
Comment (wolfs) [reply subprojects/snapshots/src/integTest/groovy/org/gradle/internal/vfs/VirtualFileSystemRetentionIntegrationTest.groovy]: Is in a separate class now.

**Comment by wolfs** [comment]
@lptr PTAL!

**Pull request merged**